frontend/customerapp/lib/models/cart.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:customerapp/models/product/product_overview.dart';
+import 'package:customerapp/models/restaurants.dart';
+import 'package:customerapp/models/user.dart';
+import 'package:flutter/widgets.dart';
+
+class Cart extends ChangeNotifier {
+  Map _order = new Map();
+
+  Cart();
+
+  void addItem(Product_overview product) {
+    _order.containsKey(product) ? _order[product]++ : _order[product] = 1;
+    notifyListeners();
+  }
+
+  void substractItem(Product_overview product) {
+    if (_order.containsKey(product)) {
+      _order[product]--;
+      if (_order[product] == 0) {
+        _order.remove(product);
+      }
+    } else {
+      throw new FormatException("No product to remove");
+    }
+    notifyListeners();
+  }
+
+  double getTotalPrice() {
+    double price = 0.0;
+    _order.forEach((k, v) => price += k.price * v);
+    return price;
+  }
+
+  Map get order {
+    return _order;
+  }
+
+  void empty() {
+    _order.clear();
+    notifyListeners();
+  }
+
+  int countItems() {
+    int numb = 0;
+    _order.forEach((k, v) => numb += v);
```

Comment:
_order.fold(0, (k, v) => t + v); (Reduce pattern :P)

Reply:
I don't get this line of code :S

Reply:
super nice code marti! good job!

Reply:
Thanks!!! :)